### PR TITLE
[feat] Authentication middleware

### PR DESCRIPTION
**Pull request by nannadao**
This PR allows to use authentication middleware to verify valid token.

### CODE REVIEW

File: src/controllers/user.ts [hunk 1]
```diff
@@ -59,7 +60,8 @@ export const googleLogin = async (
   next: NextFunction
 ) => {
   try {
-    await UserService.googleLogin(req, res)
+    const { id_token } = req.body
+    return res.json(await UserService.googleLogin(id_token, res))
   } catch (error) {
     console.log(error)
```
Comment (Rostislav013) [inline src/controllers/user.ts:66]: Handle errors. Check folder 'helpers'.

**Review by Rostislav013 — APPROVED** [review]

File: src/controllers/user.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { Request, Response, NextFunction } from 'express'
 
 import UserService from '../services/user'
 import { NotFoundError } from '../helpers/apiError'
+import { CustomRequest } from '../types'
```
Comment (MikeVedsted) [inline src/controllers/user.ts:5]: Doesn't look like you are using this here?

File: src/helpers/generateToken.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+const generateToken = (id: string) => {
+  return jwt.sign(
+    {
+      iss: 'JoinMe',
+      sub: id
+    },
+    JWT_SECRET,
+    { expiresIn: '1d' }
+  )
```
Comment (MikeVedsted) [inline src/helpers/generateToken.ts:13]: Added this for sprint review as discussed.

File: src/types.ts [hunk 4]
```diff
@@ -27,3 +29,11 @@ export type User = {
...
   created_at: Date
 }
+
+export type JwtDecoded = {
+  sub: string
+}
+
+export type CustomRequest = Request & {
+  user?: User
+}
```
Comment (MikeVedsted) [inline src/types.ts:39]: CustomRequest isn't really telling what it does etc. 
How about authenticationRequest or something along those lines?

File: src/routers/user.ts [hunk 5]
```diff
@@ -15,7 +15,7 @@ const router = express.Router()
 
 router.get('/', findAllUsers)
 router.get('/:userId', findUserById)
-router.get('/:userEmail', findUserByEmail)
+/* router.get('/:userEmail', findUserByEmail) */
```
Comment (MikeVedsted) [inline src/routers/user.ts:18]: Not sure why this is commented out? 😄

Comment (nannadao) [reply src/routers/user.ts:18]: Because they have the same endpoint, either findUserByEmail or findUserById can work. I just keep the find by ID for now

**Review by MikeVedsted — APPROVED** [review]

**Pull request merged**